File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-8.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (!root) {
            return {};
        }
        vector<vector<int>> res;
        std::queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            vector<int> level;
            int level_length = q.size();

            for (int i = 0; i < level_length; i++) {
                TreeNode* cur = q.front();
                q.pop();

                if (cur) {
                    level.push_back(cur->val);
                    q.push(cur->left);
                    q.push(cur->right);
                }
            }

            if (!level.empty()) {
                res.push_back(level);
            }
            
        }

        return res;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-8.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        res = 0

        for c in tokens:
            if c == "+":
                a, b = stack.pop(), stack.pop()
                res = b + a
                stack.append(res)

            elif c == "-":
                a, b = stack.pop(), stack.pop()
                res = b - a
                stack.append(res)

            elif c == "*":
                a, b = stack.pop(), stack.pop()
                res = b * a 
                stack.append(res)

            elif c == "/":
                a, b = stack.pop(), stack.pop()
                res = int(b / a)
                stack.append(res)

            else:
                stack.append(int(c))


        return stack[-1]
